sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -238,7 +238,7 @@ case class DropTableCommand(
     if (isTempView || catalog.tableExists(tableName)) {
       try {
         sparkSession.sharedState.cacheManager.uncacheQuery(
-          sparkSession.table(tableName), cascade = !isTempView)
+          sparkSession.table(tableName), cascade = !(isTempView && conf.storeAnalyzedPlanForView))
```

Comment:
I think we should not look at the current config value, but look at how the temp view was created. It's possible that a temp view was created with the SQL text stored, and then the config is turned on.

Reply:
Good point. Will update.